presto-main/src/main/java/com/facebook/presto/sql/planner/DistributedExecutionPlanner.java

```diff
@@ -144,11 +145,13 @@ private Visitor(Session session, StageExecutionDescriptor stageExecutionDescript
         public Map<PlanNodeId, SplitSource> visitTableScan(TableScanNode node, Void context)
         {
             // get dataSource for table
-            SplitSource splitSource = splitManager.getSplits(
+            Supplier<SplitSource> splitSourceSupplier = () -> splitManager.getSplits(
                     session,
                     node.getLayout().get(),
                     stageExecutionDescriptor.isScanGroupedExecution(node.getId()) ? GROUPED_SCHEDULING : UNGROUPED_SCHEDULING);
 
+            SplitSource splitSource = node.isTemporaryTable() ? new LazySplitSource(splitSourceSupplier::get) : splitSourceSupplier.get();
```

Comment:
nit: pass in `splitSourceSupplier` is enough. No `::get` needed :)

Reply:
It won't work because here we use Guava and in LazySplitSource we use java.util

Reply:
I think we can switch here to use java.util Supplier